Std xml utils

NotLebedev opened the pull request:
# Description

Add `xaccess`,`xupdate` and `xinsert` scripts to standard library. They allow accessing and manipulating data in new xml format https://github.com/nushell/nushell/pull/7947 with relative ease.

Access some data in nushell xml structure:
![image](https://user-images.githubusercontent.com/17511668/224785447-317359e2-1430-4dfc-9307-73f1d5e50098.png)

Update attributes of xml tags matching a path:
![image](https://user-images.githubusercontent.com/17511668/224785506-85e9aa30-b36b-43db-af1d-2f4460563124.png)


# User-Facing Changes

New commands `std xaccess`, `std xupdate` and `std xinsert`

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

> **Note**
> from `nushell` you can also use the `toolkit` as follows
> ```bash
> use toolkit.nu  # or use an `env_change` hook to activate it automatically
> toolkit check pr
> ```

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

NotLebedev commented:
Maybe we should ship this in upcoming release, otherwise there will be exactly 0 ways to work with new xml format in a sane way. This is very bare bones, but still usefulel, and I plan to add more utilities and better this two.

amtoine commented:
thanks for your contribution! :wave: 

the thing is i do not know XML at all, so i do not really know what to test and what behaviour to look after :open_mouth: 

for instance
```bash
open Cargo.toml | to xml
```
does not even work :laughing: 

in the meantime, i'll have a look at the source (but it looks already quite good and well documented, i like that :smirk:)

amtoine reviewed (changes requested):
i really do not understand the XML format, that's even the first time i see it for real i think :open_mouth: 

so i'll trust you and other XML-experts that the new `xml` module does what it's supposed to do :relieved: 

only minor remarks and i'm wondering if aligning the right operands of the `assert`s would make them a bit easier to read?
i do not know, as you feel is the best.

NotLebedev commented:
Finished this. Fixed all recommendations, added one more useful command.

Regarding 

> the thing is i do not know XML at all, so i do not really know what to test and what behaviour to look after 😮
> 
> for instance
> 
> ```shell
> open Cargo.toml | to xml
> ```
> 
> does not even work 😆

this is the reason why this commands are required at all. Xml is a weird format among others because it is not a tabular or key-value format (which are easy to use in nushell) but a strange tag-attributes-content model, allowing repetitions of entries but placing many other arbitrary restrictions. This is why `to xml` and `from xml` commands accept and produce very specific format, consisting of `record<tag attribute content>`. The help  for `to xml` and `from xml` describes it in more detail.


TL;DR xml does not map on json-like formats well, thus it can not be mapped one-to one with most nushell values in a sensible way.

amtoine commented:
@NotLebedev 

thanks for the explaination, much appreciate it :heart: 

i'm gonna review right now as it's ready :muscle:

amtoine commented on crates/nu-utils/standard_library/tests.nu:
```diff
@@ -80,7 +80,6 @@ def main [
         | flatten
         | rename file module name
     )
-
```
coul you put that line back, for readability of the test runner? :yum:

amtoine commented on test.nu:
this file does not look right :thinking: 

looks to me it should be removed, like from some intermediate test.

amtoine commented on crates/nu-utils/standard_library/test_xml.nu:
the diff of that file looks super strange, i'm not sure why it has to remove all the previous `std.nu`, put it back and then add your work :thinking:

NotLebedev replied:
I work on windows and this is the crlf bullshit that comes with it. Not sure what happened that this file converted mid-work.

amtoine reviewed (changes requested):
a few little things, we're close :yum:

fdncred commented:
@NotLebedev can we resolve the conflicts so we can land this please?

fdncred commented:
Thanks!

fdncred commented:
@NotLebedev any ideas why I'm getting errors with `xaccess` on linux?
![image](https://user-images.githubusercontent.com/343840/230420594-128192d1-c828-41d4-a006-8094338ae16c.png)

fdncred commented:
Sorry @NotLebedev - it was my fault. i had another `match` command defined.
